src/utils/http.rs

```diff
@@ -28,19 +30,24 @@ pub fn follow_redirects<F>(
     make_request: F,
 ) -> ResponseFuture<ClientResponse, SendRequestError>
 where
-    F: Fn(&str) -> ClientRequest + Send + 'static,
+    F: Fn(&str) -> Result<ClientRequest, actix_web::error::Error> + Send + 'static,
 {
     let state = (initial_url, max_redirects, true);
     Box::new(future::loop_fn(state, move |(url, redirects, trusted)| {
-        let mut request = make_request(url.as_str());
+        let mut request = tryf!(make_request(url.as_str()).map_err(|err| {
+            let fail =
+                failure::format_err!("Bad URI when creating request: {} (URI: {})", err, url);
```

Comment:
This can also be an error other than URI parsing, should we write a more generic message here?

Reply:
good point, improved